doc/uguide/cpp_api.tex

```diff
@@ -0,0 +1,223 @@
+\chapter{Include ARTS in third-party C++}
+%--------------------------------------------------------------------------
+\label{sec:cpp_api}
+
+\starthistory
+  2020-10-06 & Created and written by Richard Larsson.\\
+\stophistory
+
+%
+% Introduction
+%
+
+It is possible to include ARTS in another C++ program using
+the \verb|public_arts_interface| and linking to \verb|autoarts.h|.
+This will pull in the ARTS public API defined in \verb|agendas.cc|,
+\verb|groups.cc|, \verb|methods.cc|, \verb|workspace.cc| as well
+as an automatically generated ARTS namespace to your project.
+
+\FIXME{a private Module that imports only the ARTS namespace
+       should be added as soon as soon as C++20 becomes norm}
+
+\section{Linking the public interface}
+%--------------------------------------------------------------------------
+\label{sec:cpp_api:public_linking}
+
+To link the public interface, you need to \verb|add_subdirectory(arts)|
+anywhere in your CMake project, where the \verb|arts| directory should contain
+a current version of ARTS.  You also need to manually link OpenMP again.
+
+An example (in your projects \verb|CMakeLists.txt|):
+\begin{verbatim}
+###########################################################
+# OpenMP Library
+include (FindOpenMP)
+if (NOT ${OPENMP_FOUND})
+  message(FATAL_ERROR "No OPEN MP found, ARTS requires it")
+endif()
+
+# ARTS Library
+add_library (arts_interface STATIC interface.cpp)
+target_link_libraries(arts_interface PUBLIC
+                      public_arts_interface
+                      OpenMP::OpenMP_CXX)
```

Comment:
With PR #237 it should not be necessary anymore to call `FindOpenMP` and specify OpenMP::OpenMP_CXX when linking against public_arts_interface.

How about using `add_executable` instead of `add_library` in this example? It's probably more likely that someone wants to build an external program that links against the ARTS library than another library?